pkg/build/api/types.go

```diff
@@ -911,6 +915,9 @@ type GenericWebHookEvent struct {
 
 	// Env contains additional environment variables you want to pass into a builder container
 	Env []kapi.EnvVar
+
+	// StrategyOptions contains additional strategy-specific options for the build
+	StrategyOptions *BuildStrategyOptions
```

Comment:
yes, BinaryBuildRequestOptions and BuildLogOptions, in this file.

The nesting has value for options that are common across the strategies (if we had it to do over again, Env would probably go into BuildStrategyOptions), but i'm not going to fight unnesting it into an options field per strategy, as you suggest for the BuildRequest object below.